MavenQuickStart/quickApp/src/main/java/com/globant/topicthree/exerciseone/ShoppingCartServiceProxy.java

```diff
@@ -0,0 +1,32 @@
+package com.globant.topicthree.exerciseone;
+
+/**
+ * Proxy class for work with services.
+ * 
+ * @author andres.vaninetti
+ *
+ */
+public class ShoppingCartServiceProxy implements ShoppingCartService {
+
+	private ShoppingCartService shoppingcartUtilities;
+
+	public ShoppingCartServiceProxy(ShoppingCartService shoppingcartService) throws InterruptedException {
+		Thread.sleep(1000); // Add delay throug the proxy.
```

Comment:
Yes, that's the mainly idea. I would to change this approach of the solution, talking with Marco, understood a little more about services and have too coupled code.
I'm working in a new approach of this exercise. 
Thanks Sebas!